beater/server.go

```diff
@@ -135,34 +128,23 @@ func processRequest(
 	buf, err := ioutil.ReadAll(limitedReader)
 	if err != nil {
 		// If we run out of memory, for example
-		return reportError(500, fmt.Errorf("Data read error: %s", err))
+		return 500, err
 	}
 
 	if err = p.Validate(buf); err != nil {
-		return reportInfo(400, fmt.Errorf("Data validation error: %s", err))
+		return 400, err
 	}
 
 	list, err := p.Transform(buf)
 	if err != nil {
-		return reportInfo(400, fmt.Errorf("Data transformation error: %s", err))
+		return 400, err
 	}
 
-	responseValid.Inc()
-
 	if err = report(list); err != nil {
-		return reportError(503, fmt.Errorf("Error adding data to internal queue: %s", err))
+		return 503, err
 	}
-	return 202, nil
-}
-
-func reportError(code int, err error) (int, error) {
-	logp.Err(err.Error())
-	return code, err
-}
 
-func reportInfo(code int, err error) (int, error) {
```

Comment:
This reverts the cleanup which was done by @urso in https://github.com/elastic/apm-server/pull/118 which I quite liked.

Reply:
I don't think that is true, biggest cleanup parts are `processRequest` `sendStatus`, and `sendJSON` which don't change here. This replaces 2 functions just performing a side effect with a single line here: https://github.com/elastic/apm-server/pull/181/files#diff-f913ad38aa36e2c00a5c7f20e5ec3d20R216

Is it wrong?

Reply:
I see this as a further improvement of @urso 's nice cleanup. He cleaned up how we manage errors and extraced two functions to call. Now those two functions are called automatically when returning the response. It is a revert in the sense that those functions are removed again, but the code is not put back to where it was before, but rather enhanced the idea from urso (at least this is how I understand it).